py_gtfs_rt_ingestion/lib/convert_gtfs.py

```diff
@@ -1,76 +1,90 @@
-        essentially a small database with each contained file (outside of feed
-        info) acting as its own table. info on the gtfs scheduling standard can
-        be found at http://gtfs.org/schedule/
-        """
-        process_logger = ProcessLogger("convert_single_gtfs", filename=filename)
-        process_logger.log_start()
-
-        try:
-            tables = []
-            filelike_input: Union[str, IO[bytes]] = filename
-            last_modified = 0
-            if filename.startswith("s3://"):
-                file_to_load = str(filename).replace("s3://", "")
-                filelike_input, last_modified = get_zip_buffer(file_to_load)
-
-            with zipfile.ZipFile(filelike_input) as gtfs_zip:
-                # iterate over each "table"
-                for gtfs_filename in gtfs_zip.namelist():
-                    with gtfs_zip.open(gtfs_filename) as file:
-                        table = csv.read_csv(file)
-
-                        # add the last modified timestamp
-                        timestamp = [last_modified] * table.num_rows
-                        table = table.append_column("timestamp", [timestamp])
-
-                    filename_prefix = gtfs_filename.replace(".txt", "").upper()
-                    tables.append((filename_prefix, table))
-
-            process_logger.log_complete()
-            self.archive_files.append(filename)
-            return tables
-        except Exception as exception:
-            process_logger.log_failure(exception)
-            self.error_files.append(filename)
-            return []
+    def convert(self) -> None:
+        archive_files = []
+        error_files = []
+
+        for file in self.files:
+            process_logger = ProcessLogger("gtfs_rt_convert", filename=file)
+            process_logger.log_start()
+            try:
+                tables = zip_to_pyarrow(file)
+
+                for s3_prefix, table in tables:
+                    write_parquet_file(
+                        table=table,
+                        filetype=s3_prefix,
+                        s3_path=os.path.join(
+                            os.environ["SPRINGBOARD_BUCKET"],
+                            DEFAULT_S3_PREFIX,
+                            s3_prefix,
+                        ),
+                        partition_cols=["timestamp"],
+                    )
+
+                archive_files.append(file)
+                process_logger.log_complete()
+
+            except Exception as exception:
+                process_logger.log_failure(exception)
+                error_files.append(file)
+
+        if len(error_files) > 0:
+            move_s3_objects(
+                error_files,
+                os.path.join(os.environ["ERROR_BUCKET"], DEFAULT_S3_PREFIX),
+            )
+
+        if len(archive_files) > 0:
+            move_s3_objects(
+                archive_files,
+                os.path.join(os.environ["ARCHIVE_BUCKET"], DEFAULT_S3_PREFIX),
+            )
```

Comment:
convert the file to tables, write them, move files.

much less boilerplate by encapsulating this.